pkg/storage/redis.go

```diff
@@ -0,0 +1,216 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	goredislib "github.com/go-redis/redis/v8"
+	"github.com/go-redsync/redsync/v4"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v8"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const PONG = "PONG"
+const RedisScanBatchSize = 1000
+const RedisMutex = "redis-mutex"
+
+func init() {
+	_ = RegisterStorage(&RedisDB{})
+}
+
+type RedisDB struct {
+	db    *goredislib.Client
+	ctx   context.Context
```

Comment:
This should likely be the first parameter of all the interface methods, so we can propagate deadlines as well as cancel requests.

Reply:
Yea if this is a common golang thing, sounds good. 
So no matter what db we use mongo / bolt / couch they can all use the same context object ?

Reply:
<img width="914" alt="image" src="https://user-images.githubusercontent.com/5314059/208203409-4f3eb1f8-65c7-47a8-bae8-45154689e452.png">
You are saying to add context as the first param to all these functions?

And then many places throughout the codebase they will have to access this context, is there a good way to do this?

Reply:
Every incoming request should create a context. That should be passed along throughout the stack for any operations that do IO. Cancelations  are done using it.

Reply:
Added issue for this - https://github.com/TBD54566975/ssi-service/issues/228